src/pages/scheduler/shared/__tests__/exporter.test.ts

```diff
@@ -0,0 +1,175 @@
+import { MeetingTimes } from 'lib/fetchers';
+
+import { coursesToVCalendar, courseToVEvent } from 'pages/scheduler/shared/exporter';
+import { createVCalendar, createVEvent } from 'pages/scheduler/shared/ical';
+import { CourseCalendarEvent } from 'pages/scheduler/shared/types';
+
+const baseMeetingTime = ({
+  dateRange,
+  days,
+  time,
+}: {
+  dateRange: string;
+  days: string;
+  time: string;
+}): MeetingTimes => ({
+  dateRange,
+  days,
+  time,
+  instructors: [],
+  scheduleType: '',
+  type: 'Every Week',
+  where: 'ABC',
+});
+
+const baseEvent = (meetingTime: MeetingTimes): CourseCalendarEvent => ({
+  subject: 'CSC',
+  code: '123',
+  sectionCode: 'A01',
+  meetingTime,
+  term: '202109',
+});
+
+describe('exporters', () => {
+  describe('courseToVEvent', () => {
+    it('should return undefined when a VEvent cannot be made', () => {
+      const noTimeEvent = baseEvent(baseMeetingTime({ dateRange: '', days: '', time: 'TBA' }));
+      expect(courseToVEvent(noTimeEvent)).toBeUndefined();
+      const notEveryWeekEvent = {
```

Comment:
why is this case undefined?

Reply:
Because it can't make an event when there's no time given.